src/cpp/rtps/writer/ReaderProxy.hpp

```diff
@@ -346,6 +346,22 @@ class ReaderProxy
         return changes_low_mark_;
     }
 
+    SequenceNumber_t first_irrelevant_removed() const
+    {
+        return first_irrelevant_removed_;
+    }
+
+    SequenceNumber_t last_irrelevant_removed() const
+    {
+        return last_irrelevant_removed_;
+    }
+
+    void reset_irrelevant_removed()
+    {
+        first_irrelevant_removed_ = SequenceNumber_t::unknown();
+        last_irrelevant_removed_ = SequenceNumber_t::unknown();
+    }
+
```

Comment:
Either put the implementation of these methods in the cpp, or mark them as inline.